apps/testingaccessibility/src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,111 @@
+import NextAuth from 'next-auth'
+import GithubProvider from 'next-auth/providers/github'
+import GoogleProvider from 'next-auth/providers/google'
+import jwt from 'jsonwebtoken'
+import {JWT} from 'next-auth/jwt'
+
+export default NextAuth({
+  secret: process.env.NEXTAUTH_SECRET,
+  jwt: {
+    secret: process.env.NEXTAUTH_SECRET,
+    encode: async ({secret, token, maxAge}) => {
+      const encodedToken = jwt.sign(
+        {
+          ...token,
+          iat: Date.now() / 1000,
+          exp: Math.floor(Date.now() / 1000) + 24 * 60 * 60,
+          'https://hasura.io/jwt/claims': {
+            'x-hasura-allowed-roles': ['user'],
+            'x-hasura-default-role': 'user',
+            'x-hasura-role': 'user',
+            'x-hasura-user-id': token?.id,
+          },
```

Comment:
this is for hasura where some headers are required to be here:

https://hasura.io/docs/latest/graphql/core/auth/authentication/jwt.html#the-spec

this creates a token that you can use to make authenticated requests to hasura based on the identity supplied by a provider